File: 009_password_checker/pass_check.rb
#!/usr/bin/env ruby

ARGF.each_with_index do |line, idx|
    num     = line.count("\d{1}")
    letters = line.count("[a-zA-Z]+")

    if num < 2 or letters < 4
        puts 'INVALID'
    else
        puts 'VALID' 
    end
end

File: 009_password_checker/pass_check.py
#!/usr/bin/python

import sys
import re

for line in sys.stdin:
   num_digit = len(re.findall("\d{1}", line))
   num_letters = len(re.findall("[a-zA-Z]{1}", line))
   
   if num_digit < 2 or num_letters < 4:
     print "INVALID"
   else:
     print "VALID"
